keps/sig-cli/859-kubectl-headers/README.md

```diff
@@ -257,11 +265,161 @@ included.
 
 ### Upgrade / Downgrade Strategy
 
-NA
+Not applicable. There are no cluster components affected by this feature.
+
 
 ### Version Skew Strategy
 
-NA
+Not applicable. There is nothing required of the API Server, so there
+can be no version skew.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+
+  This feature is enabled with the KUBECTL_COMMANDS_HEADER environment
+  variable set on the client command line. Since it only affects the client
+  kubectl, it does not affect any cluster components.
+
+  - [X] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: kubectl-commands-in-headers
+    - Components depending on the feature gate: kubectl
+  - [X] Other
+    - Describe the mechanism: setting an client-side environment variable for kubectl.
+    - Will enabling / disabling the feature require downtime of the control
+      plane? No
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+      No.
+
+* **Does enabling the feature change any default behavior?**
+
+  No. This feature is not user facing, so it does not change behavior.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+
+  Yes. This feature can be disabled by simply removing an environment variable
+  on the client command line.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+
+  Re-enabling this feature is simply accomplished by setting the feature
+  environment variable on the client command line. There is no state, and there
+  is no consequence for re-enabling the feature.
+
+* **Are there any tests for feature enablement/disablement?**
+
+  There will be unit tests and integration tests which test this
+  feature enablement and disablement by setting the environment
+  variable.
+
+### Rollout, Upgrade and Rollback Planning
+
+* **How can a rollout fail? Can it impact already running workloads?**
+
+  This feature only affects kubectl; not any cluster components. So it would not
```

Comment:
the KEP doesn't talk about a max size on these headers.  A max size seems like a good idea to avoid issues.

Reply:
what value would you suggest?

Reply:
> what value would you suggest?

I'll open bidding at 4k. :)

Reply:
That is a great idea. I've added a few sentences about adding this `MAX_HEADERS` concept into the design. I've also added some more flavor to the next section `What specific metrics should inform a rollback`. Please have a look.